sql/core/src/main/scala/org/apache/spark/sql/jdbc/HiveDialect.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.jdbc
+
+import java.util.Locale
+
+import org.apache.spark.sql.catalyst.SQLConfHelper
+
+private case object HiveDialect extends JdbcDialect with SQLConfHelper {
+
+  override def canHandle(url : String): Boolean =
+    url.toLowerCase(Locale.ROOT).startsWith("jdbc:hive2")
```

Comment:
hi @dongjoon-hyun, thanks for your review. 
Your considerations are correct, but this patch is applicable to both `Hive Thrift Server` and `Spart Thrift Server`.

> You want to achieve to introduce Hive-specific syntax via this HiveDialect instead of Spark Thrift Server, right?
Actually, it's not. I used `sbin/start-thriftserver.sh` in the production environment.

Reply:
If you are trying to use this for Spark Thrift Server, this should be `SparkDialect` in Spark community. However, in that case, it will look very weird because Apache Spark **needs** a direct to access itself. That's the meaning why we don't want to add any `SparkDialect` or `HiveDialect`.
> Actually, it's not. I used sbin/start-thriftserver.sh in the production environment.

Reply:
Hi @dongjoon-hyun, we want to query data from two independent data centers, so we use multiple spark jdbc catalogs.